chat-server/src/app.ts

```diff
@@ -71,7 +37,41 @@ const reqHandler: RequestHandler = (req, _res, next) => {
   next();
 };
 
-export const setupApp = async () => {
+export const makeApp = async (): Promise<
```

Comment:
to make the app more testable, i think it'd make sense to pass mongodb, conversationService, store, dataStreamer, and embed as args to makeApp(). 

then we can pass these things in in the index.ts function which is the only file that's actually involved in running the app.